statsmodels/tsa/statespace/kalman_filter.py

```diff
@@ -0,0 +1,830 @@
+    filtered_state_cov : array
+        The filtered state covariance matrix at each time period.
+    predicted_state : array
+        The predicted state vector at each time period.
+    predicted_state_cov : array
+        The predicted state covariance matrix at each time period.
+    kalman_gain : array
+        The Kalman gain at each time period.
+    forecasts : array
+        The one-step-ahead forecasts of observations at each time period.
+    forecasts_error : array
+        The forecast errors at each time period.
+    forecasts_error_cov : array
+        The forecast error covariance matrices at each time period.
+    loglikelihood : array
+        The loglikelihood values at each time period.
+    collapsed_forecasts : array
+        If filtering using collapsed observations, stores the one-step-ahead
+        forecasts of collapsed observations at each time period.
+    collapsed_forecasts_error : array
+        If filtering using collapsed observations, stores the one-step-ahead
+        forecast errors of collapsed observations at each time period.
+    collapsed_forecasts_error_cov : array
+        If filtering using collapsed observations, stores the one-step-ahead
+        forecast error covariance matrices of collapsed observations at each
+        time period.
+    standardized_forecast_error : array
+        The standardized forecast errors
+    """
+    _filter_attributes = [
+        'filter_method', 'inversion_method', 'stability_method',
+        'conserve_memory', 'tolerance', 'loglikelihood_burn', 'converged',
+        'period_converged', 'filtered_state', 'filtered_state_cov',
+        'predicted_state', 'predicted_state_cov',
+        'forecasts', 'forecasts_error', 'forecasts_error_cov',
+        'loglikelihood'
+    ]
+
+    _filter_options = (
+        KalmanFilter.filter_methods + KalmanFilter.stability_methods +
+        KalmanFilter.inversion_methods + KalmanFilter.memory_options
+    )
+
+    _attributes = FrozenRepresentation._model_attributes + _filter_attributes
+
+    def __init__(self, model):
+        super(FilterResults, self).__init__(model)
+
+        # Setup caches for uninitialized objects
+        self._kalman_gain = None
+        self._standardized_forecasts_error = None
+
+    def update_representation(self, model, only_options=False):
+        if not only_options:
+            super(FilterResults, self).update_representation(model)
+
+        # Save the options as boolean variables
+        for name in self._filter_options:
+            setattr(self, name, getattr(model, name, None))
+
+    def update_filter(self, kalman_filter):
+        # State initialization
+        self.initial_state = np.array(
+            kalman_filter.model.initial_state, copy=True
+        )
+        self.initial_state_cov = np.array(
+            kalman_filter.model.initial_state_cov, copy=True
+        )
+
+        # Save Kalman filter parameters
+        self.filter_method = kalman_filter.filter_method
```

Comment:
All attributes are added as None in __init__ (actually performed in the base class, `FrozenRepresentation`).